tower-balance/src/load/track.rs

```diff
@@ -0,0 +1,77 @@
+use futures::{Future, Poll};
+use std::marker::PhantomData;
+
+/// Attaches `T`-typed trackers to `V` typed values.
+///
+/// This utility allows load metrics to have a protocol-agnostic means to track streams
+/// past their initial response future. For example, if `V` represents an HTTP response
+/// type, an implementaton could add `T`-typed trackers to the HTTP response extensions.
+///
+/// Trackers are intended to be RAII guards that primarily implement `Drop` and update
+/// load metric state as they are dropped.
+///
+/// A base `impl<T, V> Track<T, V> for ()` is provided that drops the tracker
+/// immediately.
+pub trait Track<T, V> {
```

Comment:
Why is this generic over `T`? Is there a reason to not pin this to `Tracker` or whatever is defined above?

Reply:
The intent is that each `Load` type may provide a different type.  pending_requests::Tracker simply holds a handle so that, when the request is dropped, the reference count is updated.

peak_ewma::Tracker will have different logic. When a `peak_ewma::Tracker` is blocked, it will grab a mutex and update the cost with the appropriate decay.

Implementations `Track` may or may not care what type of Tracker is provided.  For instance, the tower_h2_balance Track implementations don't care what type of tracker is used. It's conceivable, though, that `Load` and `Track` implementations could rely on a richer tracker type (that has more features than implementing Drop).

Reply:
Yeah, this seems good.

Reply:
Ok, this makes sense.